### DISCUSSION TITLE
Where(...).Any(...) combination not working correctly on IRavenQueryable

### DISCUSSION THREAD

**Discussion by ThierryRosseau** [ravendb/ravendb#17558] [General]
I'm currently using RavenDB 5.4.111 and encountered the following.

If i query an collection with a Where() that contains || (or) then the following Any(predicate) method wil not include parentheses.

```csharp
// Queryable

var queryable = Enumerable.Range(0, 10).AsQueryable();

var q1 = queryable.Where(s => s < 3 || s > 7).Any(s => s == 4 || s == 6); // false
var q2 = queryable.Where(s => s < 3 || s > 7).Where(s => s == 4 || s == 6).Any(); // false

Debug.Assert(q1 == q2);
```
The Queryable version returns both `false`  for q1 and q2 as expected.

```csharp
// RavenQueryable

var ravenQueryable = session.Query<Number>();

var r1 = ravenQueryable.Where(n => n.Value < 3 || n.Value > 7).Any(n => n.Value == 4 || n.Value == 6); // true

// Captured Query: from 'Numbers' where Value < $p0 or Value > $p1 and Value = $p2 or Value = $p3 limit $p4, $p5 {"p0":3,"p1":7,"p2":4,"p3":6,"p4":0,"p5":0}

var r2 = ravenQueryable.Where(n => n.Value < 3 || n.Value > 7).Where(n => n.Value == 4 || n.Value == 6).Any(); // false

// Captured Query: from 'Numbers' where (Value < $p0 or Value > $p1) and (Value = $p2 or Value = $p3) limit $p4, $p5 {"p0":3,"p1":7,"p2":4,"p3":6,"p4":0,"p5":0}

Debug.Assert(r1 == r2);
```
But the IRavenQueryable returns `true` for r1 and `false` for r2 witch is very weird.

As you can see from the captured queries the first one has no parentheses for the Where predicate and Any predicate. Therefore the logic of the operators is different and returns an incorrect result.

**Comment by ayende** [comment]
Looks like a bug, yes, see: https://issues.hibernatingrhinos.com/issue/RavenDB-21589/Parenthesis-missing-when-combining-Where-and-Any

**Comment by arekpalinski** [comment]
Fixed in https://github.com/ravendb/ravendb/pull/17568
